### PR TITLE
Fix not bound functions and methods filtering

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
I completely failed my previous PR (only tested it on sourceview, shame on me...). However I still have an issue: some types have a `ns_id` equal to 0. Therefore, the filtering is failing because the full name doesn't match (`*` instead of `Gtk` for example). For example we have this issue for:

```
Gtk.TextIter::get_marks because of *.TextMark
```

Any idea @EPashkin?

### CODE REVIEW

File: src/library.rs [hunk 1]
```diff
@@ -688,6 +688,26 @@ impl Type {
...
+
+    pub fn get_real_type<'a>(&'a self, env: &'a Env) -> &'a Type {
+        match *self {
+            Type::Array(t) |
+            Type::CArray(t) |
+            Type::FixedArray(t, _) |
+            Type::PtrArray(t) |
+            Type::List(t) |
+            Type::SList(t) => return env.type_(t).get_real_type(env),
+            _ => {}
```
Comment (EPashkin) [inline src/library.rs]: This function need return not only `&'a Type` but `t.ns_id` as well.

Comment (GuillaumeGomez) [reply src/library.rs]: Testing!

**Comment by GuillaumeGomez** [comment]
@EPashkin: All fixed, thanks a lot, wonderful idea!

File: src/library.rs [hunk 2]
```diff
@@ -688,6 +688,44 @@ impl Type {
...
+                            return inner(t, env)
+                        }
+                        _ => {}
+                    }
+                    let ns_id = t.ns_id;
+                    Some((ty, ns_id))
+                }
+                return inner(t, env);
+            }
+            _ => {}
```
Comment (EPashkin) [inline src/library.rs]: Why not?
```
            Type::SList(t) => {
                let ty = env.type_(t);
                ty.get_inner_type(env).or(Some((ty, t.ns_id)))
            }
            _ => None
```

Comment (EPashkin) [reply src/library.rs]: Although clippy want `.or_else(|| Some((ty, t.ns_id)))`

**Comment by GuillaumeGomez** [comment]
Updated.

**Comment by EPashkin** [comment]
@GuillaumeGomez Thanks

**Pull request merged**